### PR TITLE
Fix JSON decoding + Throw API Error instead of Alamofire Error when possible

### PR DESCRIPTION
**Pull request by pakko972**

### CODE REVIEW

File: Sources/APIProvider.swift [hunk 1]
```diff
@@ -8,6 +8,8 @@
 import Foundation
 import Alamofire
 
+public typealias Handler<T> = (Result<T>) -> Void
```
Comment (AvdLee) [inline Sources/APIProvider.swift]: I would rename this to `RequestCompletionHandler` to be more explicit

File: Sources/Models/AppResponse.swift [hunk 2]
```diff
@@ -29,35 +29,84 @@ public struct AppResponse: Decodable {
...
+                self = try .betaGroup(BetaGroup(from: decoder))
+            case "prereleaseVersions":
+                self = try .prereleaseVersion(PrereleaseVersion(from: decoder))
+            case "betaAppLocalizations":
+                self = try .betaAppLocalization(BetaAppLocalization(from: decoder))
+            case "builds":
+                self = try .build(Build(from: decoder))
+            case "betaLicenseAgreements":
+                self = try .betaLicenseAgreement(BetaLicenseAgreement(from: decoder))
+            case "betaAppReviewDetails":
```
Comment (AvdLee) [inline Sources/Models/AppResponse.swift]: Can we make these string values strong types? e.g. 

```swift
case .betaGroups
```

File: Sources/Models/AppResponse.swift [hunk 3]
```diff
@@ -29,35 +29,84 @@ public struct AppResponse: Decodable {
...
+            return nil
+        }
+    }
+    
+    public var betaAppReviewDetail: BetaAppReviewDetail? {
+        switch self {
+        case .betaAppReviewDetail(let value):
+            return value
+        default:
+            return nil
```
Comment (AvdLee) [inline Sources/Models/AppResponse.swift]: This adds quite some overhead. Do you think it's possible to make this more efficient in less duplicate code?

File: Sources/Models/AppsResponse.swift [hunk 4]
```diff
@@ -30,43 +30,88 @@ public struct AppsResponse: Decodable {
...
+            return nil
+        }
+    }
+    
+    public var betaAppReviewDetail: BetaAppReviewDetail? {
+        switch self {
+        case .betaAppReviewDetail(let value):
+            return value
+        default:
+            return nil
```
Comment (AvdLee) [inline Sources/Models/AppsResponse.swift]: We're creating quite some duplicate code here. Would be awesome if we can reuse logic across models!

File: Sources/Models/AppResponse.swift (hunk 2 above)
Comment (AvdLee) [reply Sources/Models/AppResponse.swift]: yes, much better!

File: Sources/APIProvider.swift (hunk 1 above)
Comment (pakko972) [reply Sources/APIProvider.swift]: Indeed

File: Sources/Models/AppResponse.swift (hunk 3 above)
Comment (pakko972) [reply Sources/Models/AppResponse.swift]: I could not come up with a nicer alternative for now, I'll remove those and maybe think about it later :)

**Comment by AvdLee** [comment]
Great work on the feedback, looks good to me now! 🎉

**Pull request merged**